subiquity/server/controllers/userdata.py

```diff
@@ -27,6 +35,25 @@ class UserdataController(NonInteractiveController):
 
     def load_autoinstall_data(self, data):
         self.model.clear()
+        # Raises cloudinit.config.schema.SchemaValidationError or
+        # SchemaDeprecationError depending on version of cloud-init to alert to
+        # cloud-config issues in provided autoinstall.user-data.
+        # cloud-init v. 22.3 will allow for log_deprecations=True to avoid
+        # raising errors on deprecated keys.
+        try:
+            validate_cloudconfig_schema(data, strict=True)
+        except SchemaValidationError as e:
+            if hasattr(e, "schema_deprecations"):
+                warnings = []
+                for schema_path, message in getattr(e, "schema_deprecations"):
```

Comment:
Per above if check I suppose we can simplify a bit
```suggestion
                for schema_path, message in e.schema_deprecations
```
Thoughts?

Reply:
And I think we can do some refactoring of this block to pull this and the similar one in subiquity/models/subiquity.py

Reply:
@dbungert refactored common code to a public method `SubiquityModel.validate_cloudconfig_schema` which the controller can call too.